posthog/api/capture.py

```diff
@@ -71,6 +73,23 @@
 )
 
 
+LIKELY_ANONYMOUS_IDS = {
+    "anon",
+    "anon_id",
+    "anonymous",
+    "anonymous_id",
+    "distinct_id",
+    "distinctid",
+    "email",
+    "false",
+    "guest",
+    "id",
+    "not_authenticated",
+    "true",
+    "undefined",
+}
```

Comment:
Cool! I'd also add: 'nan', 'none', 'null', '0', 'undefined' from the `CASE_SENSITIVE_ILLEGAL_IDS` set. No need to do a case-sensitive check on that though, we can just bundle them in as having a false-positive is less of an issue

Reply:
maybe make this an env, so we can share it across capture and ingestion? Or if not then add a comment linking to the ingestion one we likely want these to be in sync.

Reply:
There's not a big imperative to keeping these values in sync (and they'll probably diverge). But I agree that we should add a harry comment here + [link to the p-s source](https://github.com/PostHog/posthog/blob/f065757ae74c2469f0ebd0d0b5bc2c35f650c601/plugin-server/src/worker/ingestion/person-state.ts#L22-L33)

Reply:
I've added a comment for now. We can expand on this with an env variable later.